### ISSUE TITLE
hi, thank you very much for your work, you are cool

### ISSUE THREAD

**Issue by SemVolk** [Okery/PyTorch-Simple-MaskRCNN#17]
can you please give me an answer
is it possible to test this network on data that was not in the training and does not have annotations ?
I'm trying to write a prediction , but I get errors when loading images into the tenser

thank you in advance
my mail if you need: ankozoid@mail.ru

**Comment by Okery** [comment]
Can you give me your code snippet?

**Comment by SemVolk** [comment]
import torch
import matplotlib.pyplot as plt
import pytorch_mask_rcnn as pmr
from PIL import Image
from torchvision import transforms
from pytorch_mask_rcnn import visualize

use_cuda = True
dataset = "coco"
ckpt_path = "./checkall/maskrcnn_coco-100.pth"
data_dir = "./data/coco2019/"


device = torch.device("cuda" if torch.cuda.is_available() and use_cuda else "cpu")
if device.type == "cuda":
    pmr.get_gpu_prop(show=True)
print("\ndevice: {}".format(device))

ds = pmr.datasets(dataset, data_dir, 'val2019', train=True)
# indices = torch.randperm(len(ds)).tolist()
# d = torch.utils.data.Subset(ds, indices)
d = torch.utils.data.DataLoader(ds, shuffle=True)

model = pmr.maskrcnn_resnet50(True, len(ds.classes) + 1).to(device)
model.eval()
model.head.score_thresh = 0.3

if ckpt_path:
    checkpoint = torch.load(ckpt_path, map_location=device)
    model.load_state_dict(checkpoint["model"])
    print(checkpoint["eval_info"])
    del checkpoint

for p in model.parameters():
    p.requires_grad_(False)


img = Image.open("./data/coco2019/test2019/JPEGImages/3.jpg.jpg")
img.convert('RGB')

img = transforms.ToTensor()(img)

images = img.to(device)[0]


results = model(images)
plt.figure(figsize=(12, 15))
pmr.show(images, results, ds.classes)

**Comment by SemVolk** [comment]
here I'm trying to check the trained weights in 1 picture

**Comment by Okery** [comment]
There is a bug in ```images = img.to(device)[0]```
img.shape = torch.Size([3, H, W])
thus ```images = img.to(device)```

**Comment by SemVolk** [comment]
because of this, an error occurs:

RuntimeError: Error(s) in loading state_dict for MaskRCNN:
	size mismatch for head.box_predictor.cls_score.weight: copying a param with shape torch.Size([4, 1024]) from checkpoint, the shape in current model is torch.Size([14, 1024]).
	size mismatch for head.box_predictor.cls_score.bias: copying a param with shape torch.Size([4]) from checkpoint, the shape in current model is torch.Size([14]).
	size mismatch for head.box_predictor.bbox_pred.weight: copying a param with shape torch.Size([16, 1024]) from checkpoint, the shape in current model is torch.Size([56, 1024]).
	size mismatch for head.box_predictor.bbox_pred.bias: copying a param with shape torch.Size([16]) from checkpoint, the shape in current model is torch.Size([56]).
	size mismatch for head.mask_predictor.mask_fcn_logits.weight: copying a param with shape torch.Size([4, 256, 1, 1]) from checkpoint, the shape in current model is torch.Size([14, 256, 1, 1]).
	size mismatch for head.mask_predictor.mask_fcn_logits.bias: copying a param with shape torch.Size([4]) from checkpoint, the shape in current model is torch.Size([14]).

**Comment by SemVolk** [comment]
you are very cool , do you have any details for donations ?:)

**Comment by Okery** [comment]
```ckpt_path = "./checkall/maskrcnn_coco-100.pth"```
Did you train the model with COCO for 100 entire epochs?

**Comment by SemVolk** [comment]
No, it was for a small test. By the way, if I don't have a large datatset for class 1, what is the average number of epochs required? for example I have 1000 marked up images divided into train and val . how much epoch i need ?

**Comment by Okery** [comment]
If you are using the pretrained model (as default), you actually don't need a large dataset (1k~2k images). 
Several epochs may make the model work.
